packages/contracts/deploy/10_framework/00_ens.ts

```diff
@@ -45,12 +49,43 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
   const daoNode = ethers.utils.namehash(daoDomain);
   const pluginNode = ethers.utils.namehash(pluginDomain);
 
-  const daoDomainOwnerAddress = await ensRegistryContract.owner(daoNode);
+  let daoDomainOwnerAddress = await ensRegistryContract.owner(daoNode);
+
+  // node hasn't been registered yet
+  if (daoDomainOwnerAddress === ethers.constants.AddressZero) {
+    const domainSplitted = daoDomain.split('.');
+    const subdomain = domainSplitted.splice(0, 1)[0];
+    const domain = domainSplitted.join('.');
+    const tx = await ensRegistryContract.setSubnodeRecord(
+      ethers.utils.namehash(domain),
+      ethers.utils.keccak256(ethers.utils.toUtf8Bytes(subdomain)),
+      deployer,
+      ENS_PUBLIC_RESOLVERS[network.name],
```

Comment:
2 things.

1. Would be nice to have this logic whatever is inside `if` in a separate function. 
2. If the script executes `setupENS` that means we're deploying to the chain that don't have ens. This means the code in the end comes here(the line I'm mentioning this and the resolver would be invalid. I guess `setupENS` should be returning `resolver` as well and you need to be passing that `resolver` in that case.

Reply:
yep you are right. moved it to another function